In [156]:
import numpy as np
import pandas as pd

In [157]:
data=pd.read_excel('Dataset_w6.xlsx')

In [5]:
data.head()

country          Region  Surface area (km2)  \
0          Afghanistan    SouthernAsia              652864   
1              Albania  SouthernEurope               28748   
2              Algeria  NorthernAfrica             2381741   
3               Angola    MiddleAfrica             1246700   
4  Antigua and Barbuda       Caribbean                 442   

   Population in thousands (2017)  GDP per capita (current US$)  \
0                           35530                         623.2   
1                            2930                        3984.2   
2                           41318                        4154.1   
3                           29784                        4714.1   
4                             102                       14764.5   

   Agricultural production index   International trade: Imports (million US$)  
0                             125                                        3568  
1                             134                                        4669  
2                             160                                       47091  
3                             175                                        8790  
4                              88                                         491

# 1. Find out how many countries are there in the dataset.

In [86]:
len(data['country'].value_counts())

196

There are 196 countries in the data set

# 3. Find out how many countries with respect to the region are available in the dataset.

In [90]:
dat=data['Region'].value_counts()
dat

WesternAsia          18
EasternAfrica        17
Caribbean            16
WesternAfrica        15
SouthernEurope       12
SouthAmerica         12
South-easternAsia    11
EasternEurope        10
NorthernEurope       10
MiddleAfrica          9
SouthernAsia          9
CentralAmerica        8
EasternAsia           7
WesternEurope         7
SouthernAfrica        5
NorthernAfrica        5
Melanesia             5
Polynesia             5
CentralAsia           5
NorthernAmerica       4
Micronesia            4
Oceania               2
Name: Region, dtype: int64

This is the number of countries in each of the regions

# 4. On average which region has the highest

### a. GDP per capita (current US$)

In [164]:
def filt(c):
    ft=data['Region']==c
    dac=data.loc[ft,'GDP per capita (current US$)']
    avg= (sum(dac))/(len(dac))
    return avg
prev_avg=0
for i in (dat.index):
    k=filt(i)
    if(k>prev_avg):
        j=i
        av=filt(i)
        prev_avg=k
print(j,'Region has the highest average GDP per capita of',av)


NorthernAmerica Region has the highest average GDP per capita of 57659.0


### b. International trade: Imports (million US$)

In [169]:
def filt(c):
    ft=data['Region']==c
    dac=data.loc[ft,'International trade: Imports (million US$)']
    avg= (sum(dac))/(len(dac))
    return avg
prev_avg=0
for i in (dat.index):
    k=filt(i)
    if(k>prev_avg):
        j=i
        av=k
        prev_avg=k
print(j,'Region has the highest average International trade: Imports of',av,'million US$')


NorthernAmerica Region has the highest average International trade: Imports of 663552.25 million US$


# 5. Find out which region has more consistent with respect to GDP per capita (current US$)

In [168]:
def filt(c):
    ft=data['Region']==c
    dac=data.loc[ft,'GDP per capita (current US$)']
    dev= np.std(dac)
    return dev
prev_dev=10000
for i in (dat.index):
    k=filt(i)
    if(k<prev_dev):
        j=i
        prev_dev=k
print(j,'Region is more consistent with respect to GDP per capita')


WesternAfrica Region is more consistent with respect to GDP per capita
